pkg/machine/provider/platform_windows.go

```diff
@@ -43,6 +43,18 @@ func Get() (vmconfigs.VMProvider, error) {
 	}
 }
 
+func GetAll() ([]vmconfigs.VMProvider, error) {
+	providers := []vmconfigs.VMProvider{
+		new(wsl.WSLStubber),
+	}
+	if !wsl.HasAdminRights() {
+		logrus.Warn("managing hyperv machines require admin authority.")
+	} else {
```

Comment:
I like the message, though. Drop to a Debug?

Reply:
I'm concerned that people would expect reset to hard reset, including removing HyperV machines. This is actually tricky, because doing a reset the way this is implemented right now means that when reset is run without admin rights, HyperV machine configs and files will be deleted but will not be unregistered from the hypervisor, since the machine conf dirs are deleted anyway. I also considered making `podman machine reset` an admin-only command to get around this problem but that would exclude WSL users who don't have admin access.

I can drop it to a debug, but I guess there's other issues here to think about, since nothing is perfect right now due to HyperV's perm limitations.

Reply:
Updated to only warn if not -f, since I think this is important information due to dangling hyperV machines.